xtask/src/changelog.rs

```diff
@@ -0,0 +1,465 @@
+use pico_args::Arguments;
+use xshell::Shell;
+
+pub(crate) fn check_changelog(shell: Shell, mut args: Arguments) -> anyhow::Result<()> {
```

Comment:
Do we need a check to make sure your working state isn't dirty? Is this destructive to the working copy?

Reply:
We might, but I don't know if it's worth handling the complexity. Could you tell me what scenario you're trying to prevent?

Reply:
- Do changes
- "I wonder if I did the changelog right"
- cargo xtask changelog
- Where changes?
- *enters rage*

Reply:
🤔 I think that case should be handled just fine by the user not specifying the `<to_commit>` positional argument, in which case the `CHANGELOG.md` in the filesystem is used. Does that sound like it works?

To more definitively answer your question, there's no mutation of the working copy happening here.